core/src/main/java/org/jdbi/v3/core/argument/ObjectFieldArguments.java

```diff
@@ -81,7 +77,7 @@ public ObjectFieldArguments(String prefix, Object bean) {
     }
 
     @Override
-    NamedArgumentFinder getNestedArgumentFinder(Object obj) {
+    NamedArgumentFinder getNestedArgumentFinder(String name, Object obj) {
```

Comment:
I think a more effective approach would be to add a `Class<?>` parameter instead of `name`, and modify the code to tolerate `obj == null`.

When you're resolving the type of the property, you use `obj.getClass()` if not null, otherwise fall back to the provided type.